### PR TITLE
minor update

### LINKED ISSUE #33: outdated version dependency?
Hi Jon, is there a certain reason why strip-comments is still depending on the outdated version 0.8.7 of extract-comments? Can we just update this in the package.json file from 

  "dependencies": {
    "extract-comments": "^0.8.7"

to

  "dependencies": {
    "extract-comments": "^0.10.1"

?

### PR DESCRIPTION
**Pull request by tunnckoCore**
- update `extract-comments` to latest `0.10.1`
- from that update come some need for tweaks (assigning opts.safe to opts.keepProtected)
- full docs, reorganize code
- using verb 0.9, and verb configs
- the **only breaking change** is because using **`esprima` under the hood** in `extract-comments`
  + so can't parse JSON, so **3 tests fails** (marked as it.skip)
- fix some tests
- drop `should` from devDeps (was used on just two lines, so..) and `test/test.js`
- drop `require('mocha')` from `test/test.js` - not needed, make problems if you just have it globally
- switch to use `assert.strictEqual`
- eslint passing (you should change some rules, because they are dropped or renamed)
  + i just installed `eslint@2.0`, so you may don't have these warnings/errors. 
- add license banners in `test/test.js` and `index.js` - with 2014 as starting year
- resolves #33 
- need more tests for 100% coverage (_currently 92-97%, don't remember_)

**semver minor: 0.5.0**

/cc @jonschlinkert, i know you are busy, but it is little update.

### CODE REVIEW

File: .verb.md [hunk 1]
```diff
@@ -45,28 +33,9 @@ strip(commentString, { preserveNewlines: true });
...
 ***
 
 {%= include("footer") %}
-
-
-```js
-//example.strip
-console.log(strip("foo // this is a comment\n/* me too */"));
-//=> 'foo'
-```
```
Comment (jonschlinkert) [inline .verb.md]: fwiw I use a middleware to merge these examples into the code comments from `index.js`. makes the code comments cleaner. no big deal though

Comment (tunnckoCore) [inline .verb.md]: i know, but i think it make more sense to be in the code comment and to be absolutely simple and minimal. that's the idea of code comments - help. :)

**Comment by jonschlinkert** [comment]
damn, I already refactored this locally, I just haven't had time to finish with trying to get verb and assemble done.  sometimes it's a good practice to mention something before putting the work into a pr like this, since it's more than a minor update. I do appreciate it though. 

I'll push my refactor to a branch, then we can discuss.

File: .verb.md (hunk 1 above)
Comment (jonschlinkert) [inline .verb.md]: I personally don't like big code comments. but you're right, simplicity wins

**Comment by tunnckoCore** [comment]
> sometimes it's a good practice to mention something before putting the work into a pr like this

yea, but always forgot to separate the things in separate commits, which would look better.

**Comment by jonschlinkert** [comment]
btw, try doing this `verb --config=tasks:readme`. let me know if that works for you correctly

**Comment by tunnckoCore** [comment]
yea yea, works, i'm just using global config from yesterday :)

**Comment by jonschlinkert** [comment]
thanks, I just went ahead and reviewed your code and published, since I'm crunched for time. It looks good, thank you

**Comment by jonschlinkert** [comment]
> yea yea, works,

good, thanks!

**Comment by tunnckoCore** [comment]
> thank you

me too.

**Pull request merged**